src/stirling/source_connectors/perf_profiler/perf_profile_connector.cc

```diff
@@ -354,23 +360,42 @@ void PerfProfileConnector::ProcessBPFStackTraces(ConnectorContext* ctx, DataTabl
   // Read BPF stack traces & histogram, build records, incorporate records to data table.
   CreateRecords(stack_traces.get(), ctx, data_table);
 
+  uint64_t num_stack_traces_sampled;
+  profiler_state_->get_value(sample_count_idx, num_stack_traces_sampled);
+  CheckProfilerState(num_stack_traces_sampled);
+
   // Now that we've consumed the data, reset the sample count in BPF.
   profiler_state_->update_value(sample_count_idx, 0);
 }
 
-void PerfProfileConnector::CheckProfilerState() {
+void PerfProfileConnector::CheckProfilerState(const uint64_t num_stack_traces) {
   uint64_t error_code;
   profiler_state_->get_value(kErrorStatusIdx, error_code);
 
   DCHECK_EQ(error_code, kPerfProfilerStatusOk);
 
   switch (error_code) {
-    case kOverflowError:
+    case kOverflowError: {
+      // overflow_ratio is actual:expected. That is, the actual number of stack traces sampled
+      // vs. the expected number of stack traces.
+      const double overflow_ratio =
+          static_cast<double>(num_stack_traces) / static_cast<double>(expected_stack_traces_);
+      profiler_state_overflow_gauge_.Set(overflow_ratio);
```

Comment:
Our metric scraping interval was changed to 15 mins (https://github.com/pixie-io/pixie/pull/1358), so we will only record the value that happens during that instant. Even with the previous 1 minute interval, there is still a risk that the gauge value fluctuates between sampling and that we are blind to its variance (possible to miss a really high, brief value).

What if we recorded the worst case overflow rather than each iteration's overflow instead? While that doesn't inform us how frequently it happens, I think that's a good indicator for if the perf profiler is performing well or not.